src/WebJobs.Script.WebHost/Management/InstanceManager.cs

```diff
@@ -372,25 +375,25 @@ private async Task<string> Download(RunFromPackageContext pkgContext)
             if (pkgContext.PackageContentLength != null && pkgContext.PackageContentLength > 100 * 1024 * 1024)
             {
                 _logger.LogInformation($"Downloading zip contents from '{cleanedUrl}' using aria2c'");
-                AriaDownload(tmpPath, fileName, zipUri);
+                AriaDownload(tmpPath, fileName, zipUri, pkgContext.IsWarmUpRequest);
             }
             else
             {
                 _logger.LogInformation($"Downloading zip contents from '{cleanedUrl}' using httpclient'");
-                await HttpClientDownload(filePath, zipUri);
+                await HttpClientDownload(filePath, zipUri, pkgContext.IsWarmUpRequest);
             }
 
             return filePath;
         }
 
-        private async Task HttpClientDownload(string filePath, Uri zipUri)
+        private async Task HttpClientDownload(string filePath, Uri zipUri, bool isWarmupRequest)
         {
             HttpResponseMessage response = null;
             await Utility.InvokeWithRetriesAsync(async () =>
             {
                 try
                 {
-                    using (_metricsLogger.LatencyEvent(MetricEventNames.LinuxContainerSpecializationZipDownload))
+                    using (_metricsLogger.LatencyEvent(isWarmupRequest ? MetricEventNames.LinuxContainerSpecializationZipDownloadWarmup : MetricEventNames.LinuxContainerSpecializationZipDownload))
```

Comment:
nit: maybe hoist this one and the below to a local like you did in ValidateBlobPackageContext